jmclient/jmclient/blockchaininterface.py

```diff
@@ -83,7 +89,7 @@ def add_tx_notify(self, txd, unconfirmfun, confirmfun, notifyaddr,
             for outs in txd['outs']:
                 addr = btc.script_to_address(outs['script'], vb)
                 try:
-                    if self.rpc('getaccount', [addr]) != '':
+                    if self.is_address_imported(addr):
```

Comment:
instead of potentially two rpc calls, why don't we just ask the wallet if the address/script belongs to us?

Reply:
How is this done?

Reply:
I'd tend to think we need to treat the backend Core instance as authoritative? I wouldn't want to rely on some logic in the code that indicates that the wallet's opinion about whether the address is imported or not is watertight. Or did I misunderstand?

Reply:
I see your point. Using an unimported address from the wallet should never happen but we have no way to make absolutely sure. otoh, if it belongs to our wallet and is not imported we should import it.

Reply:
Oh I just realized, @undeath probably meant to ask the joinmarket wallet, not the Core wallet.